spec/features/pictures_feature_spec.rb

```diff
@@ -0,0 +1,154 @@
+
+    scenario 'pictures can be viewed' do
+      visit '/pictures/new'
+      fill_in 'Description', with: 'My first picture!'
+      page.attach_file('Image', Rails.root + 'spec/factories/test.jpg')
+      click_button 'Post'
+      visit '/'
+      expect(page).to have_xpath("//img[contains(@src, \"medium/test.jpg\")]")
+    end
+
+    scenario 'picture description can be viewed' do
+      picture = FactoryGirl.create(:picture, user: FactoryGirl.create(:user, :second))
+      visit '/'
+      expect(page).to have_content(picture.description)
+    end
+
+    # As a user
+    # So that I can show that I posted a picture
+    # I would like my username to be displayed with a picture I posted
+    scenario 'username is displayed' do
+      picture = FactoryGirl.create(:picture, user: FactoryGirl.create(:user, :second))
+      visit '/pictures'
+      expect(page).to have_link picture.user.username
+    end
+  end
+
+  # As a user
+  # So that I can remove one of my posted pictures
+  # I would like to delete a picture
+  context 'deleting pictures' do
+    context 'when signed in' do
+      before(:each) do
+        user = FactoryGirl.create(:user, :first)
+        sign_in(user.email, user.password)
+      end
+
+      scenario 'users can delete their own pictures' do
+        post_picture
+        visit '/pictures'
+        click_link 'Delete'
+        expect(page).not_to have_xpath("//img[contains(@src, \"medium/test.jpg\")]")
+      end
+
+      scenario 'user cannot delete others users\' pictures' do
+        post_picture
+        click_button 'Sign out'
+        user = FactoryGirl.create(:user, :second)
+        sign_in(user.email, user.password)
+        visit '/pictures'
+        click_link 'Delete'
+        expect(page).to have_content 'You cannot delete this picture as you did not create it'
+      end
+    end
+
+    context 'when signed out' do
+      scenario 'signed out users cannot delete pictures' do
+        picture = FactoryGirl.create(:picture, user: FactoryGirl.create(:user, :second))
+        visit '/pictures'
+        expect(page).not_to have_link 'Delete'
+      end
+    end
+  end
+
+  context 'editing pictures' do
+    before(:each) do
+      user = FactoryGirl.create(:user, :first)
+      sign_in(user.email, user.password)
+    end
+
+    context 'when signed in' do
+      scenario 'user can edit their own pictues' do
+        post_picture
+        visit '/pictures'
+        click_link 'Edit'
+        fill_in 'Description', with: 'My first edited picture!'
+        click_button 'Update'
+        expect(page).to have_content 'My first edited picture!'
+      end
+
+      scenario 'user cannot edit other users\' pictures' do
+        post_picture
+        click_button 'Sign out'
+        user = FactoryGirl.create(:user, :second)
+        sign_in(user.email, user.password)
+        visit '/pictures'
+        click_link 'Edit'
+        expect(page).to have_content 'You cannot edit this picture as you did not create it'
```

Comment:
Metrics/LineLength: Line is too long. [92/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)